src/tests/JIT/SIMD/ShiftOperations.cs

```diff
@@ -0,0 +1,170 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System;
+using System.Runtime.CompilerServices;
+using System.Numerics;
+using System.Runtime.InteropServices;
+using System.Collections.Generic;
+
+public class Test
+{
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static ulong Shlx64bit(ulong x, int y) => x << y;
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static long Sarx64bit(long x, int y) => x >> y;
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static ulong Shrx64bit(ulong x, int y) => x >> y;
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static unsafe ulong ShrxRef64bit(ulong* x, int y) => *x >> y;
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static ulong Rorx(ulong x) => BitOperations.RotateRight(x, 2);
+
+    public static unsafe int Main()
+    {
+        const int PASS = 100;
+        const int FAIL = 101;
+        int returnCode = PASS;
+
+        try
+        {
+            ulong valULong = 0;
```

Comment:
I would define these variables closer to their usage.

Reply:
Fixed.